### ISSUE TITLE
strptime() panics with invalid strings

### ISSUE THREAD

**Issue by rakbladsvalsen** [pola-rs/polars#4847]
### Polars version checks

- [x] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of polars.

### Issue Description

The error is pretty simple and very straightforward to reproduce:

1. Create a simple dataframe containing a column with invalid dates. A good example might be `20205050`.
2. Parse the dates on that column with `.str().strptime()` with a valid format, i.e. `%Y` . 
3. Evil `panic!` appears! The caller doesn't get a chance to catch this error, since no `PolarsError` is returned in the result. Just a plain panic.

This panic is caused [here](https://github.com/pola-rs/polars/blob/master/polars/polars-time/src/chunkedarray/utf8/strptime.rs#L112). Basically, polars is using `parse_fmt` instead of the non-panicking `parse_fmt_opt`. 

Note that this bug is fatal Python, since `pyo3`'s runtime exceptions cannot be handled with the good ol' `exception BlahBlahException`. If this is expected behavior (panic!-ing when an invalid date is passed), then this would be a Python-only bug, since this panic literally kills the python interpreter.

### Reproducible Example
```
use polars;
use polars::prelude::*;
use std::default::Default;

use color_eyre::{Result};
fn main() -> Result<()> { 
    let df = df! [
        "a"   => ["20202020"]
    ]?;
    let options = StrpTimeOptions{
        fmt: Some("%Y".into()),
        strict: true,
        exact: true,
        ..Default::default()
    };
    let _dates = df.clone().lazy()
    .select([
        col("a").str().strptime(options)
    ])
    .collect()?;
    Ok(())
}
```

### Expected Behavior

The call to strptime() should return the appropriate `PolarsError` with the appropiate error, probably something similar to pandas' "cannot parse the remainder" or "invalid date format". 

### Installed Versions

<details>
```
[dependencies]
color-eyre = "0.6.2"
polars = { version = "0.23.2", features = ["lazy", "strings"] }
```

Full `cargo tree` output: https://termbin.com/hig0
</details>

**Comment by ritchie46** [comment]
I believe the panic originates in chrono. I tried to make this fallible a few weeks ago and hit a wall on the chrono project. I am not entirely sure anymore, but I really think it was not only our native parser that was the culprit.

**Comment by rakbladsvalsen** [comment]
> ...hit a wall on the chrono project. I am not entirely sure anymore, but I really think it was not only our native parser that was the culprit.

Agreed. 
However... it's not really chrono's fault. Chrono provides two methods to parse dates (and hms-like dates): parse_ymd and parse_ymd_opt. The first one panics if the passed values are invalid, while the second one provides a nicer `Option<>` interface.

I did some digging and this effectively prevents panics when parsing invalid dates/strings:
`Some(NaiveDate::from_ymd_opt(year, month, day)?.and_hms_nano_opt(hour, min, sec, nano)?)`

Additionally, I found out that even if the passed dates are valid (i.e. something like `fmt="%Y"` and `dates = ["2020", "2021"]`), the whole thing will fail (but not panic), because of the following defaults:
```
    let mut month: u32 = 0;
    let mut day: u32 = 0;
```
... which are invalid values for chrono.

Changing this to _month, day = 1_ completely solves all issues, though.
